src/allmydata/test/test_node.py

```diff
@@ -584,15 +606,30 @@ def test_listen_on_zero(self):
             str(ctx.exception),
         )
 
-    def test_multiple_ports(self):
+    # Randomly allocate a couple distinct port numbers to try out.  The test
+    # never actually binds these port numbers so we don't care if they're "in
+    # use" on the system or not.  We just want a couple distinct values we can
+    # check expected results against.
+    @given(ports=sets(elements=port_numbers(), min_size=2, max_size=2))
+    # A few examples should satisfy us here.  The logic for dealing with these
+    # numbers is hardly complex.  On the flip side, all of the setup work we
+    # do to get a node we can test against is pretty expensive.
+    @settings(max_examples=10)
```

Comment:
mostly curiosity, but I'm not sure I understand the motivation here? I mean: why not `max_examples=10` for every test?

Reply:
Yea I'm not 100% sure either.  I did write this part of the change in 2018 and I definitely don't remember specifically what I was thinking here.  It might have been something like this.

I don't want to hard code a port number and the behavior of the implementation shouldn't depend on the value of the port number anyway.  So I'll use Hypothesis.  Oh no this test is so slow using Hypothesis because instantiating a `Node` is an I/O and compute-heavy operation.  I guess I'll keep some value from the Hypothesis usage but crank down the iterations so it doesn't take so long.  That way I still don't have to hard-code anything and the test will still get to run against some different values but the test will also still be fast.

That said, _after_ I introduced `@settings(max_examples=10)` it looks like I changed the test some more to remove the `Node` instantiation completely... And the test seems pretty fast now, with or without this setting, so I'll probably just delete it.

That leads a bit towards an answer to the question "Why not do this everywhere?" - if the test is fast enough with the higher default number of examples, go ahead and run it that many times because we will find problems sooner.